File: backend/schemas/task.py
from pydantic import BaseModel, Field
from typing import List, Optional, Dict

class TaskCreate(BaseModel):
    title: str
    dueDate: str
    submitType: str
    keywords: List[str]
    summary: Optional[str] = None
    priority: str
    status: str = "todo"
    notificationSettings: List[int] = Field(default_factory=lambda: [60, 1440])
    notifiedOffsets: List[int] = Field(default_factory=list)

class TaskUpdate(BaseModel):
    title: Optional[str] = None
    dueDate: Optional[str] = None
    submitType: Optional[str] = None
    keywords: Optional[List[str]] = None
    summary: Optional[str] = None
    priority: Optional[str] = None
    status: Optional[str] = None
    notificationSettings: Optional[List[int]] = None

class TaskStatusUpdate(BaseModel):
    status: Optional[str] = None
    dueDate: Optional[str] = None

File: frontend/src/types/task.ts
export type TaskPriority = "high" | "medium" | "low";

export type TaskStatus = "todo" | "done";

export type Task = {
  id: string;
  title: string;
  dueDate: string;
  submitType: string;
  keywords: string[];
  summary?: string;
  priority?: TaskPriority;
  status: TaskStatus;
  createdAt?: string;
};

export type CreateTaskInput = {
  title: string;
  dueDate: string;
  submitType: string;
  keywords: string[];
  summary?: string;
  priority?: TaskPriority;
  status?: TaskStatus;
};

export type UpdateTaskInput = Partial<
  Pick<
    Task,
    "title" | "dueDate" | "submitType" | "keywords" | "summary" | "priority" | "status"
  >
>;

export type AnalyzeTaskResult = {
  title: string;
  dueDate: string;
  submitType: string;
  keywords: string[];
  summary?: string;
  priority?: TaskPriority;
};
